packages/nouns-subgraph/src/nouns-dao.ts

```diff
@@ -29,63 +32,69 @@ import {
 
   // Check if the proposer was a delegate already accounted for, if not we should log an error
   // since it shouldn't be possible for a delegate to propose anything without first being 'created'
-  if (proposer == null) {
+  if (proposer == null && parsedProposal.signers.length == 0) {
     log.error('Delegate {} not found on ProposalCreated. tx_hash: {}', [
-      event.params.proposer.toHexString(),
-      event.transaction.hash.toHexString(),
+      parsedProposal.proposer,
+      parsedProposal.txHash,
     ]);
   }
-  // Create it anyway since we will want to account for this event data, even though it should've never happened
-  proposer = getOrCreateDelegate(event.params.proposer.toHexString());
+
+  // Create it anyway, which supports V3 cases of proposers not having any Nouns
+  proposer = getOrCreateDelegate(parsedProposal.proposer);
   proposal.proposer = proposer.id;
-  proposal.targets = changetype<Bytes[]>(event.params.targets);
-  proposal.values = event.params.values;
-  proposal.signatures = event.params.signatures;
-  proposal.calldatas = event.params.calldatas;
-  proposal.createdTimestamp = event.block.timestamp;
-  proposal.createdBlock = event.block.number;
-  proposal.createdTransactionHash = event.transaction.hash;
-  proposal.startBlock = event.params.startBlock;
-  proposal.endBlock = event.params.endBlock;
-  proposal.proposalThreshold = event.params.proposalThreshold;
-  proposal.quorumVotes = event.params.quorumVotes;
+  proposal.targets = parsedProposal.targets;
+  proposal.values = parsedProposal.values;
+  proposal.signatures = parsedProposal.signatures;
+  proposal.calldatas = parsedProposal.calldatas;
+  proposal.createdTimestamp = parsedProposal.createdTimestamp;
+  proposal.createdBlock = parsedProposal.createdBlock;
+  proposal.createdTransactionHash = parsedProposal.createdTransactionHash;
+  proposal.startBlock = parsedProposal.startBlock;
+  proposal.endBlock = parsedProposal.endBlock;
+  proposal.proposalThreshold = parsedProposal.proposalThreshold;
+  proposal.quorumVotes = parsedProposal.quorumVotes;
   proposal.forVotes = BIGINT_ZERO;
   proposal.againstVotes = BIGINT_ZERO;
   proposal.abstainVotes = BIGINT_ZERO;
-  proposal.description = event.params.description.split('\\n').join('\n'); // The Graph's AssemblyScript version does not support string.replace
-  proposal.title = extractTitle(proposal.description);
-  proposal.status = event.block.number >= proposal.startBlock ? STATUS_ACTIVE : STATUS_PENDING;
+  proposal.description = parsedProposal.description;
+  proposal.title = parsedProposal.title;
+  proposal.status = parsedProposal.status;
+
+  const signerDelegates = new Array<string>(parsedProposal.signers.length);
+  for (let i = 0; i < parsedProposal.signers.length; i++) {
+    const signerAddress = parsedProposal.signers[i];
+    const signerDelegate = getOrCreateDelegateWithNullOption(signerAddress, false);
+
+    if (signerDelegate == null) {
```

Comment:
@eladmallel in which case would `signerDelegate` be `null`? shouldn't it always be not null because it's returned from `getOrCreate`?

Reply:
actually it might happen, since we don't really enforce that every single signer on a prop have at least one vote.. we actually let signers with zero votes take part.

btw, we also allow people with zero votes to cast a vote :)

key question:
should we enforce that all signers on a proposal have at least one vote?

if we say that we should enforce:
should we enforce the same thing when casting votes? and if not, what's the difference?

if we aren't enforcing:
should subgraph create a Delegate entity for them as we do in the propose and vote handlers? 
or should we leave the code as-is and not create a new delegate for such signers?

wdyt @davidbrai?

Reply:
in the meantime I'm updating the code to create a zero votes delegate for any signer without a delegate entity, to at least be consistent with how we do things in other places here. still important to discuss these questions together!

Reply:
@eladmallel what I meant is that the call your making to `getOrCreateDelegateWithNullOption` seems to always return a non-null object:
```
if (delegate == null && createIfNotFound) {
    delegate = new Delegate(id);
    ...
    return delegate;
}
```

so if I'm not mistaken, the `if (signerDelegate == null) {` check is never true

Reply:
@davidbrai I heard you, and tried making it better in this commit: https://github.com/nounsDAO/nouns-monorepo/pull/677/commits/984a3cc1cb81eb15b6363c60dfb2753230b62cb7

wdyt?

Reply:
lgtm